index.js

```diff
@@ -1,11 +1,11 @@
 // taken from https://github.com/taskcluster/taskcluster-queue/blob/0dd2b45b860836a2ef6c651c5499c9abfb5670ad/utils/aws-sdk-promise.js
 var aws = require('aws-sdk');
-var Promise = require('promise');
 
 // XXX: This is a terrible hack but it should not break anything in horrible
 // ways.
 aws.Request.prototype.promise = function() {
-  return new Promise(function(accept, reject) {
+  var Prom = aws.Promise || Promise || require('promise');
```

Comment:
In talking to @jhford on irc... This will cause `ReferenceError` when `Promise` isn't defined...

We should probably just drop the `|| Promise ||` part and fallback to `require('promise')`.